core/trino-main/src/main/java/io/trino/operator/scalar/MathFunctions.java

```diff
@@ -858,6 +864,11 @@ public static long roundReal(@SqlType(StandardTypes.REAL) long num, @SqlType(Sta
         if (rescaledRound != Long.MAX_VALUE) {
             result = sign * (rescaledRound / factor);
         }
+        else if (rescaled == Double.POSITIVE_INFINITY) {
```

Comment:
This should be `if (Double.isInfinite(rescaled))`, as should the other checks in other places.

Reply:
updated